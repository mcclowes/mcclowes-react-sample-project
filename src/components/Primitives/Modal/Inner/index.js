import Inner from "./Inner";

export default Inner;
