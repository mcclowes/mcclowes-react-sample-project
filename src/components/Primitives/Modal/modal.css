.modal-overlay {
  align-items: center;
  background-color: rgba(50, 50, 50, 0.75);
  bottom: 0px;
  display: flex;
  justify-content: center;
  left: 0px;
  right: 0px;
  top: 0px;
  transition: opacity 200ms ease-in-out 0s;
  z-index: 1000;
  padding: 1em;
  position: fixed;
  transition: opacity 200ms ease-in-out;

  opacity: 1;
  animation: fade 0.25s linear;
}

@keyframes fade {
  0% {
    opacity: 0;
  }
  100% {
    opacity: 1;
  }
}

@media only screen and (max-width: 768px) {
  .modal-overlay {
    align-items: flex-end;
  }
}

.modal-wrapper {
  align-items: center;
  background: none;
  border-radius: 5px;
  border: none;
  bottom: auto;
  display: flex;
  left: auto;
  max-height: 90vh;
  max-width: 90vw;
  outline: none;
  overflow: hidden;
  padding: 0;
  position: relative;
  right: auto;
  top: auto;
}

@media only screen and (max-width: 768px) {
  .modal-wrapper {
    min-width: 100%;
  }
}
