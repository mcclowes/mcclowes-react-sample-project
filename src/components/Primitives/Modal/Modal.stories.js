import React, { useState } from "react";
import Modal from "./Modal";
import Button from "../Button";
import Padded from "../Padded";

export default { title: "Modal" };

export const simple = () => {
  return (
    <Modal open>
      <Modal.Inner>Hello world</Modal.Inner>
    </Modal>
  );
};

export const structured = () => {
  return (
    <Modal open closeIcon doClose={() => {}}>
      <Modal.Inner>
        <Modal.Header>Header</Modal.Header>

        <Modal.Content>
          Praesent commodo cursus magna, vel scelerisque nisl consectetur et.
          Aenean eu leo quam. Pellentesque ornare sem lacinia quam venenatis
          vestibulum. Nulla vitae elit libero, a pharetra augue. Duis mollis,
          est non commodo luctus, nisi erat porttitor ligula, eget lacinia odio
          sem nec elit. Cras mattis consectetur purus sit amet fermentum.
          Vestibulum id ligula porta felis euismod semper. Lorem ipsum dolor sit
          amet, consectetur adipiscing elit.
        </Modal.Content>

        <Modal.Actions>
          <Padded space="p3">
            <Button subtle={true} text="Close" />
          </Padded>
          <Padded space="p3">
            <Button text="Submit" />
          </Padded>
        </Modal.Actions>
      </Modal.Inner>
    </Modal>
  );
};

const ControlledView = () => {
  const [open, setOpen] = useState(false);
  return (
    <Modal
      trigger={<Button text="Open Modal" doClick={() => setOpen(true)} />}
      open={open}
      closeIcon
      doClose={() => setOpen(false)}
    >
      <Modal.Inner>
        <Modal.Header>Header</Modal.Header>

        <Modal.Content>
          Praesent commodo cursus magna, vel scelerisque nisl consectetur et.
          Aenean eu leo quam. Pellentesque ornare sem lacinia quam venenatis
          vestibulum. Nulla vitae elit libero, a pharetra augue. Duis mollis,
          est non commodo luctus, nisi erat porttitor ligula, eget lacinia odio
          sem nec elit. Cras mattis consectetur purus sit amet fermentum.
          Vestibulum id ligula porta felis euismod semper. Lorem ipsum dolor sit
          amet, consectetur adipiscing elit.
        </Modal.Content>

        <Modal.Actions>
          <Padded space="p3">
            <Button doClick={() => setOpen(false)} subtle={true} text="Close" />
          </Padded>
          <Padded space="p3">
            <Button doClick={() => setOpen(false)} text="Submit" />
          </Padded>
        </Modal.Actions>
      </Modal.Inner>
    </Modal>
  );
};

export const controlled = () => {
  return <ControlledView />;
};
