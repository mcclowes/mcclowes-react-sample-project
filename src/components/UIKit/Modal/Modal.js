import PropTypes from "prop-types";
import React from "react";
import ReactModal from "react-modal";

import Actions from "./Actions";
import Close from "./Close";
import Content from "./Content";
import Header from "./Header";
import Inner from "./Inner";

import "./modal.css";

if (process.env.NODE_ENV !== "test") ReactModal.setAppElement("#root");

const Modal = (props) => {
  const { children, doClose, open, closeIcon, trigger } = props;

  return (
    <>
      {trigger && trigger}

      <div onClick={(e) => e.stopPropagation()}>
        <ReactModal
          isOpen={open}
          closeTimeoutMS={400}
          shouldCloseOnEsc={true}
          shouldCloseOnOverlayClick={true}
          onRequestClose={doClose}
          className="modal-wrapper"
          overlayClassName="modal-overlay"
        >
          {closeIcon && doClose && <Close onClick={doClose} />}

          {children}
        </ReactModal>
      </div>
    </>
  );
};

Modal.Actions = Actions;
Modal.Close = Close;
Modal.Content = Content;
Modal.Header = Header;
Modal.Inner = Inner;

Modal.propTypes = {
  children: PropTypes.node,
  closeIcon: PropTypes.bool,
  doClose: PropTypes.func,
  open: PropTypes.bool,
  trigger: PropTypes.any,
};

export default Modal;
