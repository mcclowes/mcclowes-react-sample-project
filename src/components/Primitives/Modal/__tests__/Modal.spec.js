import Modal from "../Modal";
import React from "react";
import ReactModal from "react-modal";
import { shallow } from "enzyme";

const ContentMock = () => {
  return <div>This is in a portal</div>;
};

const shallowComponent = (props = {}) => {
  const { open = true, handleClose = () => {}, closeIcon = true } = props;

  return shallow(
    <Modal open={open} doClose={handleClose} closeIcon={closeIcon}>
      <ContentMock />
    </Modal>
  );
};

describe("Modal", () => {
  it("renders open", () => {
    const wrapper = shallowComponent();

    expect(wrapper).toMatchSnapshot();
  });

  it("renders closed", () => {
    const wrapper = shallowComponent({ open: false });

    expect(wrapper).toMatchSnapshot();
  });

  it("no icon", () => {
    const wrapper = shallowComponent({ closeIcon: false });

    expect(wrapper).toMatchSnapshot();
  });
});
