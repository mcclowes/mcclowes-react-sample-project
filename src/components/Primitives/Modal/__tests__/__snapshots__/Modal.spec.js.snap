// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal no icon 1`] = `
<Fragment>
  <div
    onClick={[Function]}
  >
    <Modal
      ariaHideApp={true}
      bodyOpenClassName="ReactModal__Body--open"
      className="modal-wrapper"
      closeTimeoutMS={400}
      isOpen={true}
      onRequestClose={[Function]}
      overlayClassName="modal-overlay"
      parentSelector={[Function]}
      portalClassName="ReactModalPortal"
      role="dialog"
      shouldCloseOnEsc={true}
      shouldCloseOnOverlayClick={true}
      shouldFocusAfterRender={true}
      shouldReturnFocusAfterClose={true}
    >
      <ContentMock />
    </Modal>
  </div>
</Fragment>
`;

exports[`Modal renders closed 1`] = `
<Fragment>
  <div
    onClick={[Function]}
  >
    <Modal
      ariaHideApp={true}
      bodyOpenClassName="ReactModal__Body--open"
      className="modal-wrapper"
      closeTimeoutMS={400}
      isOpen={false}
      onRequestClose={[Function]}
      overlayClassName="modal-overlay"
      parentSelector={[Function]}
      portalClassName="ReactModalPortal"
      role="dialog"
      shouldCloseOnEsc={true}
      shouldCloseOnOverlayClick={true}
      shouldFocusAfterRender={true}
      shouldReturnFocusAfterClose={true}
    >
      <Close
        onClick={[Function]}
      />
      <ContentMock />
    </Modal>
  </div>
</Fragment>
`;

exports[`Modal renders open 1`] = `
<Fragment>
  <div
    onClick={[Function]}
  >
    <Modal
      ariaHideApp={true}
      bodyOpenClassName="ReactModal__Body--open"
      className="modal-wrapper"
      closeTimeoutMS={400}
      isOpen={true}
      onRequestClose={[Function]}
      overlayClassName="modal-overlay"
      parentSelector={[Function]}
      portalClassName="ReactModalPortal"
      role="dialog"
      shouldCloseOnEsc={true}
      shouldCloseOnOverlayClick={true}
      shouldFocusAfterRender={true}
      shouldReturnFocusAfterClose={true}
    >
      <Close
        onClick={[Function]}
      />
      <ContentMock />
    </Modal>
  </div>
</Fragment>
`;
